proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+    "properties": {
+        "chainID": {
+            "dataType": "bytes",
+            "length": CHAIN_ID_LENGTH,
+            "fieldNumber": 1
+        }
+    }
+}
+```
+
+#### removeSupportAllNFTsFromChain
+
+This event has `name = EVENT_NAME_REMOVE_SUPPORT_ALL_NFTS_FROM_CHAIN`, and is emitted when the [`removeSupportAllNftsFromChain`](#removesupportallnftsfromchain-1) function is called.
+
+##### Topics
+
+- `chainID`: the ID of the chain for which all NFTs are supported.
+
+##### Data
+
+Same as in previous event, i.e., follow the `supportAllNFTsFromChainEventDataSchema`.
+
+#### supportNftID
+
+This event has `name = EVENT_NAME_SUPPORT_NFTID`, and is emitted when the [supportNftID](#supportnftid-1) function is called.
+
+##### Topics
+
+- `nftID`: the nftID of the supported NFT.
+
+##### Data
+
+```java
+supportNftIDEventDataSchema = {
+    "type": "object",
+    "required" = ["nftID"],
+    "properties": {
+        "nftID": {
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
+            "fieldNumber": 1
+        }
+    }
+}
+```
+
+#### removeSupportNftID
 
-This function returns the locking status of an NFT.
+This event has `name = EVENT_NAME_REMOVE_SUPPORT_NFTID`, and is emitted when the [removeSupportNftID](#removesupportnftid-1) function is called.
+
+##### Topics
+
+- `nftID`: the nftID for which the support is removed.
+
+##### Data
+
+Same as in previous event, i.e., follow the `supportNftIDEventDataSchema`.
+
+### Internal Functions
+
+#### createNFTEntry
 
 ```python
-getLockingModuleID(nftID):
-    nftID = canonicalNFTID(nftID)
-    if NFTStore(nftID) exists:
-        return NFTStore(nftID).lockingModuleID
-    else:
-        return entry does not exist
+def createNFTEntry(
+    address: Address,
+    nftID: NFTID
+) -> None:
+
+    # This may fail if there is not enough fee left.
+    Fee.payFee(NFT_ENTRY_INITIALIZATION_FEE)
+
+    create substore entry with
+        substorePrefix = SUBSTORE_PREFIX_NFT
+        key = nftID
+        value = encode(
+            schema = NFTStoreSchema,
+            object = {
+               "owner": address,
+               "attributesArray": []
+            }
+        )
 ```
 
-#### getNFTowner
+#### createUserEntry
 
-This function returns the owner of an NFT.
+```python
+def createUserEntry(
+    address: Address,
+    nftID: NFTID
+) -> None:
+
+    # This may fail if there is not enough fee left.
+    Fee.payFee(USER_ENTRY_INITIALIZATION_FEE)
+
+    create substore entry with
+        substorePrefix = SUBSTORE_PREFIX_USER
+        key = address + nftID
+        value = encode(
+            schema = userStoreSchema,
+            object = {
+               "lockingModule": NFT_NOT_LOCKED
+            }
+        )
+```
+
+#### createEscrowEntry
+
+```python
+def createEscrowEntry(
+    receivingChainID: ChainID,
+    nftID: NFTID
+) -> None:
+
+    # This may fail if there is not enough fee left.
+    Fee.payFee(ESCROW_ENTRY_INITIALIZATION_FEE)
```

Comment:
I am not sure that we should pay a fee here. If the NFT is transferred cross-chain, we create an escrow entry but we also delete a user entry, so the state does not increase in size.